File: testsuite/tests/callback/minor_named_.c
#include <caml/mlvalues.h>
#include <caml/memory.h>
#include <caml/callback.h>

value incr_ref(value unit) {
  static const value* v;
  if (!v) v = caml_named_value("incr_ref");
  caml_modify(&Field(*v, 0), Val_int(Int_val(Field(*v, 0)) + 1));
  return Val_unit;
}

File: testsuite/tests/callback/minor_named.ml
(* TEST
 modules = "minor_named_.c";
*)

(* Tests Callback.register and caml_named_value on a young object *)

external incr_ref : unit -> unit = "incr_ref"

let () =
  let r = ref 40 in
  Callback.register "incr_ref" r;
  incr_ref ();
  Gc.minor ();
  incr_ref ();
  Printf.printf "%d\n" !r
